### ISSUE TITLE
Responsive breaks after adding element

### ISSUE THREAD

**Issue by mclarentgp** [STRML/react-grid-layout#247]
I am experience an issue where the responsive grid layout fails whenever I am adding a new element to this.state.layout.  I get the same issue in the demo example https://strml.github.io/react-grid-layout/examples/6-dynamic-add-remove.html.  The demo works on initial load, but the second I click the add button to add a new element and try to resize my browser window the responsive breaks.  Is this expected behavior or am I missing something on this example?

**Comment by STRML** [comment]
Definitely a bug. I'll look into it.

**Comment by mclarentgp** [comment]
I did some research into this and it appears the issues might be related only to the demo and not RGL as a whole.  I made the following changes in the demo jsx file to allow the responsive grid to work when adding a new element.  

`items: {lg: [0, 1, 2, 3, 4].map(function(i, key, list) {
            return {i: i.toString(), x: i * 2, y: 0, w: 2, h: 2, add: i === (list.length - 1).toString()};
        })},`
this was done in order to get the function findOrGenerateResponsiveLayout in responsiveUtils.js to work correctly for the responsiveness of the layout.  In doing this change I found that I needed to have the verticalCompact property set to true at all times so that the Y coordinate on the added element would be able to get changed from Infinity to a number.

These last two changes were done so that when the new element was added the y coordinate would be updated from Infinity to a number during the function getFirstCollision in utils.js
`  onLayoutChange(layout, items) {
    this.props.onLayoutChange(layout, items);
  },`
moved {...this.props} as the first attribute in ResponsiveReactGridLayout 
`<ResponsiveReactGridLayout {...this.props} onLayoutChange={this.onLayoutChange} onBreakpointChange={this.onBreakpointChange} ` as it would seem that if this is not done onLayoutChange does not seem to trigger.

Let me know what your thoughts are on the changes if they seem viable or if im way off on this.  I've included the entire jsx demo code as it contains additional changes that were the result of one or some of the changes mentioned above.
[dynamic-add-remove.txt](https://github.com/STRML/react-grid-layout/files/313895/dynamic-add-remove.txt)
